tests/handlers/github_test.py

```diff
@@ -0,0 +1,151 @@
+"""Test the github webhook handler."""
+
+import hashlib
+import hmac
+from dataclasses import dataclass
+from pathlib import Path
+from string import Template
+
+import pytest
+import respx
+from httpx import AsyncClient
+
+from ..support.constants import TEST_GITHUB_WEBHOOK_SECRET
+from ..support.gafaelfawr import mock_gafaelfawr
+
+
+@dataclass(frozen=True)
+class GithubRequest:
+    payload: str
+    headers: dict[str, str]
+
+
+def webhook_request(org: str, repo: str, ref: str) -> GithubRequest:
+    """Build a Github webhook request and headers with the right hash."""
+    data_path = Path(__file__).parent.parent / "data"
+    template = (data_path / "github_webhook.tmpl.json").read_text()
+    payload = Template(template).substitute(
+        org=org,
+        repo=repo,
+        ref=ref,
```

Comment:
Super-minor nit: I think this may all fit on one line if you delete the trailing comma after ref and let Ruff try to reformat it.